packages/es-indexer/src/listener.ts

```diff
@@ -53,64 +53,71 @@ export function takePending() {
-    }
-  },
-  reposts: (repost: RepostRow) => {
-    pending.reposts.push(repost)
-    if (repost.repost_type == 'track') {
-      pending.trackIds.add(repost.repost_item_id)
-    } else {
-      pending.playlistIds.add(repost.repost_item_id)
-    }
-  },
-  follows: (follow: FollowRow) => {
-    pending.follows.push(follow)
-    // followee follower_count comes from aggregate_user
-    // which is update async...
-    // marking followee_user_id stale here is likely to result in a noop
-    // as aggregate_user hasn't been updated yet...
-    // so instead we listen for update on that table to ensure follower_count gets updated.
-    // pending.userIds.add(follow.followee_user_id)
-    pending.userIds.add(follow.follower_user_id)
-  },
-  users: (user: UserRow) => {
-    pending.userIds.add(user.user_id)
-  },
-  tracks: (track: TrackRow) => {
-    pending.trackIds.add(track.track_id)
-  },
-  playlists: (playlist: PlaylistRow) => {
-    pending.playlistIds.add(playlist.playlist_id)
-  },
-}
-
 export async function startListener() {
   const connectionString = process.env.audius_db_url
   const client = new Client({ connectionString, application_name: 'es-listen' })
   await client.connect()
   const tables = LISTEN_TABLES
   const sql = tables.map((t) => `LISTEN ${t}; `).join(' ')
 
-  client.on('notification', (msg) => {
+  const handlers = {
+    aggregate_user: (row: AggregateUserRow) => {
+      pending.userIds.add(row.user_id)
+    },
+    aggregate_plays: (row: AggregatePlayRow) => {
+      if (!row.play_item_id) return // when could this happen?
+      pending.trackIds.add(row.play_item_id)
+    },
+    // TODO: can we do trigger on agg playlist matview?
+    saves: (save: SaveRow) => {
+      pending.saves.push(save)
+      if (save.save_type == 'track') {
+        pending.trackIds.add(save.save_item_id)
+      } else {
+        pending.playlistIds.add(save.save_item_id)
+      }
+    },
+    reposts: (repost: RepostRow) => {
+      pending.reposts.push(repost)
+      if (repost.repost_type == 'track') {
+        pending.trackIds.add(repost.repost_item_id)
+      } else {
+        pending.playlistIds.add(repost.repost_item_id)
+      }
+    },
+    follows: (follow: FollowRow) => {
+      pending.follows.push(follow)
+      // followee follower_count comes from aggregate_user
+      // which is update async...
+      // marking followee_user_id stale here is likely to result in a noop
+      // as aggregate_user hasn't been updated yet...
+      // so instead we listen for update on that table to ensure follower_count gets updated.
+      // pending.userIds.add(follow.followee_user_id)
+      pending.userIds.add(follow.follower_user_id)
+    },
+    users: (user: UserRow) => {
+      pending.userIds.add(user.user_id)
+    },
+    tracks: async (track: TrackRow) => {
+      pending.trackIds.add(track.track_id)
+      // re-index any playlists containing this track
+      const playlists = await client.query(
```

Comment:
awesome, could you also use the `playlists_containing_tracks` column on track? avoids a query potentially.

Reply:
I realized... postgres notify has a size limit of 8kb... so for larger rows (users, playlists, tracks) we just send the ID... so in current setup we have to do a followup query in `listener.ts` at least.

Another option would be to go to `TrackIndexer.ts` and use the `playlists_containing_tracks` to mark playlist IDs as pending...  will give that a try.